rcl/include/rcl/log_level.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__LOG_LEVEL_H_
+#define RCL__LOG_LEVEL_H_
+
+#include "rcl/allocator.h"
+#include "rcl/macros.h"
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/// A logger item to specify a name and log_level
+typedef struct rcl_logger_setting_t
+{
+  /// name for the logger.
+  char * name;
+  /// level for the logger.
+  int level;
+} rcl_logger_setting_t;
+
+/// Hold log_level.
+typedef struct rcl_log_level_t
+{
+  /// Default log level (represented by `RCUTILS_LOG_SEVERITY` enum) or -1 if not specified.
```

Comment:
If this can only be vales in an enum, why not using that type?
Unfortunately, the type was names with the wrong casing, but adding a typedef in rcl sounds like a good idea:
```c
typedef RCUTILS_LOG_SEVERITY rcl_log_severity_t;
```

we could also rename `severity` to `level`

Reply:
> I think the reason why using int directly is that the rcutils_logging_set_logger_level uses 'int' as an argument.

mmm, I didn't see that. Ok, using `int` directly sounds good.